File: internal/model/song.go
package model

import (
	"time"
)

// Song - 音乐文件元数据
type Song struct {
	ID        uint      `gorm:"primaryKey"`
	FilePath  string    `gorm:"uniqueIndex;not null"`
	Folder    string    `gorm:"index"` // 文件夹路径，用于分组浏览
	Title     string
	Artist    string
	Album     string
	Year      int
	Genre     string
	TrackNum  int
	Duration  int // 秒
	CoverPath string
	Lyrics    string
	FileHash  string    `gorm:"index"`
	FileSize  int64
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: frontend/src/types/song.ts
// Song - 歌曲类型定义
export interface Song {
  id: number;
  filePath: string;
  title: string;
  artist: string;
  album: string;
  year: number;
  genre: string;
  trackNum: number;
  duration: number;
  coverPath: string;
  lyrics: string;
  fileHash: string;
  fileSize: number;
  createdAt: string;
  updatedAt: string;
}

// ArtistWithCount - 艺术家及其歌曲数量
export interface ArtistWithCount {
  id: number;
  name: string;
  songCount: number;
}

// AlbumWithCount - 专辑及其歌曲数量
export interface AlbumWithCount {
  id: number;
  name: string;
  artist: string;
  songCount: number;
}
